test/spec/data/action.js

```diff
@@ -0,0 +1,90 @@
+module.exports = {
+  name: 'basis.data.action',
+  init: function(){
+    var basis = window.basis.createSandbox();
+    var DataObject = basis.require('basis.data').Object;
+    var STATE = basis.require('basis.data').STATE;
+    var createAction = basis.require('basis.data.action').create;
+    var resolvedFn = function(){
+      return Promise.resolve('success');
+    };
+    var rejectedFn = function(){
+      return Promise.reject('fail');
+    };
+  },
+  test: [
+    {
+      name: 'Action from promise',
+      test: [
+        {
+          name: 'Should call every callback when fulfilled',
+          test: function(done){
+            var passed = {};
+
+            createAction({
+              fn: resolvedFn,
+              start: function(){
+                assert(this instanceof DataObject);
+                passed.start = true;
+              },
+              success: function(data){
+                assert(this instanceof DataObject);
+                assert(String(this.state) == STATE.PROCESSING);
+                assert(data == 'success');
+                passed.success = true;
+              },
+              failure: function(){
+                assert('', 'Failure callback must not be called!');
```

Comment:
I think we should add asset.failure() or assert.wrongInvocation() for those cases.

```
assert(false, 'Failure callback must not be called!');
```